packages/file-search/src/browser/FileQuery.ts

```diff
@@ -0,0 +1,52 @@
+/********************************************************************************
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Position } from '@theia/editor/lib/browser';
+import { integer } from '@theia/core/shared/vscode-languageserver-types';
+import { injectable } from '@theia/core/shared/inversify';
+
+@injectable()
+export class FileQuery {
+    private _filePattern: string = '';
+    private lineNumber: integer = 0;
+
+    /**
+     * Parses the query in its different components and returns the corresponding file pattern of the query
+     * e.g. /path/to/file:10, will parse the query and return /path/to/file.
+     *
+     * @param fileQuery a file query in the form "filePathPattern:location number".
+     *
+     * @returns the filePathPattern portion of the query.
+     */
+    parseFileQuery(fileQuery: string): string {
+        const queryParts = fileQuery.split(':', 2);
+        this._filePattern = queryParts[0].trim();
+
+        const locationString = queryParts.length > 1 ? queryParts[1].trim() : '';
+        const locationNumber = Number.parseInt(locationString);
+        this.lineNumber = locationNumber && locationNumber > 1 ? locationNumber - 1 : 0;
+
+        return this.filePattern;
+    }
```

Comment:
This `FileQuery` component is a bit of overhead just to parse queries.

Rather you could move this `parseFileQuery` back into `QuickFileOpenService` as a protected method and make it return a struct like `{ file: string, position: string }`.

Reply:
I have addressed this re-using an interface used in vscode for this purpose, 
I have also re-used vscode's regex pattern so we now parse the line and column.